src/lib/tools.js

```diff
@@ -0,0 +1,10 @@
+import olLayerGroup from 'ol/layer/Group.js'
+
+export default function traverseLayer(layer, ancestors, visitor) {
```

Comment:
indeed, that's a good idea, I also thought about it. But this function is only needed with OL 6.5, so it will be dropped when transitioning to maplibre and OL>6.15 in v4. I think it's clearer when the function is grouped with the custom Mapbox lib